CIE-5514 - Sensitive endpoints to be blocked (#3980)
SoftwareAG/c8y-docs: content/edge/edge-configuration-bundle/configuring-nginx.md

Reviewer: This will also need a release note referring to this, pointing out that it's actually something important

Author: Sure, but this is just a doc change and no actual release, would it still be ok to add a release note since saran is already preparing a communication which includes this too.

Reviewer: You can have changelogs/release notes that aren't a product change, but just an "Announcement", which would fit here. Having it as an announcement makes it clear that the upcoming communication to customers definitely applies even to Edge.

Author: Taken note of this suggestion and will be implemented for later scenarios, this, however, is an exception and is excluded from the release notes for this instance. Thank you @rthrippleton

---
weight: 50

> title: Configuring Nginx

Reviewer 2: ```suggestion
title: Configuring NGINX
```

layout: bundle
sector:
  - edge_server
---

### Blocking Sensitive Endpoints in NGINX

> To prevent access to the /service/<service>/heapdump endpoint, create an NGINX location rule that returns 403 Forbidden for all heapdump requests.

Reviewer 3: ```suggestion
To prevent access to the `/service/<service>/heapdump` endpoint, create an NGINX location rule that returns "403 Forbidden" for all heapdump requests.
```

---
weight: 50
title: Configuring NGINX
layout: bundle
sector:
  - edge_server
---

### Blocking sensitive endpoints in NGINX

To block access to specific endpoints, create an NGINX location rule that returns "403 Forbidden" for matching requests. You can use this approach to disable any endpoint by configuring the appropriate location pattern.

{{< c8y-admon-important >}}

Perform these operations as `root` user.

{{< /c8y-admon-important >}}

The following example shows how to block a specific endpoint. You can modify the location pattern to match any endpoint you want to disable:

```shell

> CONFIG_FILE="/etc/nginx/conf.d/blocked_endpoints.common.location.conf"

Reviewer 3: Does the file name have any significance? If not, can we rename it to blocked-endpoints.conf?

Author: Significance in the way we are importing the files into post.nginx.conf file.

```
http {
    include preamble.http.conf;

    server {
        include http_listen.conf;
        include conf.d/*.common.location.conf;
        include conf.d/*.http.location.conf;

        # Temporary (hopefully) backward compatibility for DataHub
        include conf.d/common/*.location.conf;
    }

    server {
        include https_listen.conf;
        include conf.d/*.common.location.conf;
        include conf.d/*.https.location.conf;
        include conf.d/installation/*.location.post.conf;
    }

    include conf.d/*.http.conf;
}

```

Any combination works here as long as the regex can parse the new file name.

Reviewer 3: ok, then, let us name the file as `blocked-endpoints.common.location.conf`

Author: Done, changed as suggested

cat >> "$CONFIG_FILE" <<'EOF'

# Add your exact regex inside the location directive
# Example: 

> # location ~* ^/service/[^/]+/<endpoint>$ {

Reviewer 3: Could you put an example service endpoint in this?

Author: Done, Added few examples

---
weight: 50
title: Configuring NGINX
layout: bundle
sector:
  - edge_server
---

### Blocking sensitive endpoints in NGINX

> To block access to specific endpoints, create an NGINX location rule that returns "403 Forbidden" for matching requests. You can use this approach to disable any endpoint by configuring the appropriate location pattern.

Reviewer 3: ```suggestion
Configure an NGINX `location` block to return a `403 Forbidden` status code for all matching incoming requests. This method can be used to effectively disable access to an endpoint with a defined URL pattern or to blacklist specific users within that location block.
```

{{< c8y-admon-important >}}

Perform these operations as `root` user.

{{< /c8y-admon-important >}}

The following example shows how to block a specific endpoint. You can modify the location pattern to match any endpoint you want to disable:

```shell
CONFIG_FILE="/etc/nginx/conf.d/blocked-endpoints.common.location.conf"

cat >> "$CONFIG_FILE" <<'EOF'

# Add your exact regex inside the location directive
# Example1: Blocks health endpoint for all services
# location ~* ^/service/[^/]+/health$ {
#     return 403;
# }
# Example2: Blocks health endpoint only for cep service
# location ~* ^/service/cep/health$ {
#     return 403;
# }

location ~* <your-regex-here> {
    return 403;
}
EOF

chown nginx:nginx "$CONFIG_FILE"
chmod 644 "$CONFIG_FILE"
systemctl restart nginx
```

```shell
CONFIG_FILE="/etc/nginx/conf.d/blocked-endpoints.common.location.conf"

cat >> "$CONFIG_FILE" <<'EOF'

# Add your exact regex inside the location directive

> # Example1: Blocks health endpoint only for cep service

Reviewer 3: Update this comment as it is pointing to the endpoint not the http authorization header.
